matron/src/device/device_monitor.c

```diff
@@ -276,15 +277,25 @@ void add_dev_tty(struct udev_device *dev) {
     char *name = get_device_name(dev);
     if (fnmatch("/dev/ttyUSB*", node, 0) == 0) {
         fprintf(stderr, "dev_monitor: got ttyUSB, assuming grid\n");
-        dev_list_add(DEV_TYPE_MONOME, node, name);
+        dev_list_add(DEV_TYPE_MONOME, node, name, NULL);
     } else if (is_dev_monome_grid(dev)) {
         fprintf(stderr, "dev_monitor: TTY appears to be ACM grid\n");
-        dev_list_add(DEV_TYPE_MONOME, node, name);
+        dev_list_add(DEV_TYPE_MONOME, node, name, NULL);
     } else if (is_dev_crow(dev)) {
         fprintf(stderr, "tty is a crow\n");
-        dev_list_add(DEV_TYPE_CROW, node, name);
+        dev_list_add(DEV_TYPE_CROW, node, name, NULL);
     } else {
         fprintf(stderr, "dev_monitor: unmatched TTY device %s at %s\n", name, node);
+        const char *vendor, *model;
+        vendor = udev_device_get_property_value(dev, "ID_VENDOR");
+        model = udev_device_get_property_value(dev, "ID_MODEL");
```

Comment:
Reiterating serial-number comment from previously, and remembering some serial devices have multiple channels. Can't remember if this has to be selected via a _channel_ or _index_ with udev.

Reply:
Hmm, it would be helpful to have an example of this. I added the interface number from `ID_USB_INTERFACE_NUM` to the serial config args. Based on some googling it seems like devices with multiple serial ports might use that to distinguish between ports? Let me know if that seems right to you!

Reply:
Note sure! I have a 4-channel FTDI module that I can test with to verify.